packages/@lwc/babel-plugin-component/src/decorators/wire/transform.js

```diff
@@ -153,14 +153,16 @@ function buildWireConfigValue(t, wiredValues) {
             }
 
             if (wiredValue.params) {
+                const dynamicParamNames = wiredValue.params.map(p => {
+                    return t.stringLiteral(
```

Comment:
The [objectProperty](https://babeljs.io/docs/en/babel-types#objectproperty) can't be an expression, therefore is an `Identifier` | `Literal`, and where the value (param name) comes from is different.